src/CleanSoftwareCron.php

```diff
@@ -191,13 +191,23 @@ protected static function deleteItems(
     ): int {
         global $DB;
 
-        $scope['LIMIT'] = min($max, self::MAX_BATCH_SIZE);
+        $total = 0;
+
+        do {
+
+            $scope['LIMIT'] = ($max > self::MAX_BATCH_SIZE) ? self::MAX_BATCH_SIZE : $max;
```

Comment:
I think it could be simplified to this:
```suggestion
            $scope['LIMIT'] = min($max, self::MAX_BATCH_SIZE)
```